appinventor/components/src/com/google/appinventor/components/runtime/SoundRecorder.java

```diff
@@ -62,9 +80,17 @@
       recorder.setOnInfoListener(SoundRecorder.this);
     }
 
-    void start() {
+    void start() throws IllegalStateException {
       Log.i(TAG, "starting");
+      try {
       recorder.start();
+      } catch (IllegalStateException e) {
+        // This is the error produced when there are two recorders running.
+        // There might be other causes, but we don't know them.
+        Log.i(TAG, "got IllegalStateException. Are there two recorders running?");
```

Comment:
Why don't you use Log.e(). It will let you provide the exception as its third argument which will log the stack trace to the system log.